monkey/infection_monkey/network_scanning/mssql_fingerprinter.py

```diff
@@ -77,27 +80,50 @@ def _query_mssql_for_instance_data(host: str) -> Optional[bytes]:
         sock.close()
 
 
-def _get_services_from_server_data(data: bytes) -> Dict[str, Any]:
-    services = {MSSQL_SERVICE: {}}
-    services[MSSQL_SERVICE]["display_name"] = DISPLAY_NAME
-    services[MSSQL_SERVICE]["port"] = SQL_BROWSER_DEFAULT_PORT
+def _get_instance_info(instance_info: Sequence[str]) -> Dict[str, str]:
+    info = {}
+    it = iter(instance_info)
+    for k in it:
+        info[k] = next(it)
```

Comment:
Never modify the data structure you are traversing! https://dev.to/mcsee/code-smell-111-modifying-collections-while-traversing-5aac

Reply:
How are we modifying the date structure that we are traversing?!
`info` is a dictionary is not the one we tranverse.